cheroot/ssl/builtin.py

```diff
@@ -102,12 +137,44 @@ def __init__(
         if self.ciphers is not None:
             self.context.set_ciphers(ciphers)
 
+        self._server_env = self.env_cert_dict(
+            'SSL_SERVER', self.parse_cert(certificate, private_key),
+        )
+        if not self._server_env:
+            return
+        cert = None
+        with open(certificate, mode='rt') as f:
+            cert = f.read()
+
+        # strip off any keys by only taking the first certificate
+        cert_start = cert.find(ssl.PEM_HEADER)
+        if cert_start == -1:
+            return
+        cert_end = cert.find(ssl.PEM_FOOTER, cert_start)
+        if cert_end == -1:
+            return
+        cert_end += len(ssl.PEM_FOOTER)
+        self._server_env['SSL_SERVER_CERT'] = cert[cert_start:cert_end]
+
     def bind(self, sock):
         """Wrap and return the given socket."""
         return super(BuiltinSSLAdapter, self).bind(sock)
 
+    @staticmethod
+    def sni_callback(sock, sni, context):
```

Comment:
Let's move this to a separate function too. I don't see any need for keeping it in the class.

Reply:
moved